File: PlayToLearn/app/src/main/java/com/avoid/playtolearn/model/Board.java
package com.avoid.playtolearn.model;

import android.content.Context;

import com.avoid.playtolearn.common.Global;
import com.avoid.playtolearn.game.GameLogic;

import java.io.Serializable;
import java.util.ArrayList;

public class Board implements Serializable{
    private int currentX = 0;
    private int currentY = 0;

    private ArrayList<ArrayList<BoardTile>> tileGrid = new ArrayList<>();

    public Board(Context context) {
        initializeBoard();
    }

    public void initializeBoard(){
        for(int col = 0; col < Global.COLUMN_COUNT; col++){
            ArrayList<BoardTile> tileColumn = new ArrayList<>();
            for(int row = 0; row < Global.ROW_COUNT; row++){
                BoardTile boardTile = new BoardTile();
                boardTile.setRow(row);
                boardTile.setColumn(col);

                if(col == currentX && row == currentY){
                    boardTile.setBoardTileState(BoardTileState.CURRENT);
                    boardTile.setQuestion(GameLogic.getRandomQuestion());
                }
                tileColumn.add(boardTile);
            }
            tileGrid.add(tileColumn);
        }
    }

    public ArrayList<ArrayList<BoardTile>> getTileGrid() {
        return tileGrid;
    }

    public void setTileGrid(ArrayList<ArrayList<BoardTile>> tileGrid) {
        this.tileGrid = tileGrid;
    }

    public BoardTile getBoardTile(int row, int column){
        return getTileGrid().get(column).get(row);
    }
}

File: app/src/main/java/com/ivantha/playtolearn/model/Board.kt
package com.ivantha.playtolearn.model

class Board {

    companion object {
        const val COLUMN_COUNT = 6
        const val ROW_COUNT = 10
    }

    var tileGrid = ArrayList<ArrayList<Tile>>()
    var currentX = 0
    var currentY = 0

    constructor()

    init {
        for (col in 0 until COLUMN_COUNT) {
            val tileColumn = ArrayList<Tile>()
            for (row in 0 until ROW_COUNT) {
                val tile = Tile()
                tile.row = row
                tile.column = col

                var q = Question()
                q.description = "asdasd"
                q.title = "asdasd"
                q.answer = 4
                tile.question = q

                tileColumn.add(tile)
            }
            tileGrid.add(tileColumn)
        }

        // Set (0, 0) as the current tile
        tileGrid[0][0].boardTileState = Tile.BoardTileState.CURRENT
    }
}
